### PR TITLE
Fix/docker install

### PR DESCRIPTION
**Pull request by sgilliam15**

### CODE REVIEW

File: example.env [hunk 1]
```diff
@@ -4,8 +4,16 @@ SERVER_PORT=4000
 # Slack
```
Comment (StephRicardo) [inline example.env:4]: To keep the naming consistent, let's use this. I don't think we'll be using those other ENVs. 

```
SLACK_BOT_OAUTH_TOKEN
SLACK_BOT_USER_ID
SLACK_BOT_CLIENT_ID
SLACK_BOT_CLIENT_SECRET
SLACK_BOT_CLIENT_SIGNING_SECRET
SLACK_BOT_REDIRECT_URL
SLACK_TEAM_ID
SLACK_BOT_VERIFICATION_TOKEN
```

File: src/bots/slack/adapter.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    botToken: process.env.SLACK_BOT_TOKEN,
+
+    // parameters used to secure webhook endpoint
+    verificationToken: process.env.SLACK_BOT_VERIFICATION_TOKEN,
+    clientSigningSecret: process.env.SLACK_CLIENT_SIGNING_SECRET,
+
+    // credentials used to set up oauth for multi-team apps
+    clientId: process.env.SLACK_CLIENT_ID,
+    clientSecret: process.env.SLACK_CLIENT_SECRET,
+    scopes: ['bot'],
```
Comment (StephRicardo) [inline src/bots/slack/adapter.ts:57]: For the install endpoint to work. we need to change the scopes. The bot scope is a legacy one. 

I added a bunch of possible scopes here: https://github.com/TrimAgency/sprint-bot/pull/1/files#diff-461a9672fe884beb7c4d6097795f4cbf

I also recommend we use `oauthVersion: 'v2'`

File: src/bots/slack/controller.ts [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    webhook_uri: SLACK_WEBHOOK,
+    storage,
+    // ...other options
+  });
+
+  controller.ready(async () => {
+    // Generic message response
+    onMessageConversation(controller);
+
+    // On DM example
```
Comment (StephRicardo) [inline src/bots/slack/controller.ts:21]: Could we add a note to remove all these examples? Alternatively, we can extract these out to example files that can be removed.

File: src/server.ts [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
-// Server Setup
-const port = process.env.SERVER_PORT || 4000;
+// Connect to DB via Mongoose
+db();
+
+// Initialize Slackbot
+initSlackBot();
+
+// Express Server Setup
+const port = process.env.EXPRESS_SERVER_PORT || 4000;
```
Comment (StephRicardo) [inline src/server.ts:23]: FYI, the PORT env is used by Botkit to edit what port we want it to use. We can't run both servers on this port.

File: src/server.ts [hunk 5]
```diff
@@ -1,18 +1,28 @@
-import { Express } from 'express';
-import * as express from 'express';
+// Load enviornment
```
Comment (StephRicardo) [inline src/server.ts:1]: `// Load environment` 😄

File: src/constants.ts [hunk 6]
```diff
@@ -1,2 +1,5 @@
 // Mongo
-export const MONGO_URI = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@bot-db:27017/${process.env.DATABASE_NAME}`;
+export const MONGO_URI = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@bot-db:27017/?authSource=admin`;
```
Comment (StephRicardo) [inline src/constants.ts:2]: Is docker working?

File: src/bots/slack/conversations/message.conversation.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { SlackBotWorker } from 'botbuilder-adapter-slack';
+import { BotkitMessage, Botkit } from 'botkit';
+
+export const onMessageConversation = async (controller: Botkit) => {
+  controller.on(
+    'message',
+    async (bot: SlackBotWorker, message: BotkitMessage) => {
+      await bot.reply(message, 'I heard a message!');
```
Comment (StephRicardo) [inline src/bots/slack/conversations/message.conversation.ts:8]: Do we want to add a helpful message here? In the coffee bot, it responds to all random words with

```
I don't know what you mean. Type `order` to get started!"
```

File: src/bots/slack/conversations/message.conversation.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { SlackBotWorker } from 'botbuilder-adapter-slack';
```
Comment (StephRicardo) [inline src/bots/slack/conversations/message.conversation.ts:1]: I think we should finalize the filing for this during the meeting. `conversations` is helpful, but excludes other slack features we may add with botkit, like slash_commands and files for interactive messages. 

We'll need to think about how we want to organize large conversations and related files.

File: src/bots/slack/controller.ts (hunk 3 above)
Comment (sgilliam15) [reply src/bots/slack/controller.ts:21]: I'm extracting them.  and removing the one we don't want right now

File: src/server.ts (hunk 4 above)
Comment (sgilliam15) [reply src/server.ts:23]: botkit is running on 3000 by default.  I think it is just a wrapper for express

File: src/server.ts (hunk 5 above)
Comment (sgilliam15) [reply src/server.ts:1]: I spell that wrong 90% of the time

File: src/constants.ts (hunk 6 above)
Comment (sgilliam15) [reply src/constants.ts:2]: yes.  I've updated the readme

File: src/bots/slack/conversations/message.conversation.ts (hunk 7 above)
Comment (sgilliam15) [reply src/bots/slack/conversations/message.conversation.ts:8]: I'll make another PR to revise the convo's.

File: src/bots/slack/conversations/message.conversation.ts (hunk 8 above)
Comment (sgilliam15) [reply src/bots/slack/conversations/message.conversation.ts:1]: ok, yes let's

**Pull request merged**